src/components/report-technical-issue/template.njk

```diff
@@ -1,8 +1,14 @@
 {% set language %}{% if params.language == 'cy' %}cy{% else %}en{% endif %}{% endset %}
 {% set serviceId = params.serviceId or params.serviceCode %}
+{% set urlParametersBuilder %}
+  {% if serviceId %}?service={{ serviceId | urlencode }}{% endif %}
+  {% if params.referrerUrl and serviceId %}&amp;{% endif %}
+  {% if params.referrerUrl and not serviceId %}?{% endif %}
```

Comment:
Technically, I guess this could be the first part, where we just look for `serviceId` or `params.referrerUrl`.